waiter/test/waiter/auth/oauth_test.clj

```diff
@@ -0,0 +1,211 @@
+;;
+;;       Copyright (c) 2017 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.auth.oauth-test
+  (:require [clojure.core.async :as async]
+            [clojure.data.codec.base64 :as b64]
+            [clojure.data.json :as json]
+            [clojure.string :as str]
+            [clojure.test :refer :all]
+            [full.async :as fa]
+            [qbits.jet.client.http :as http]
+            [waiter.auth.authentication :as auth]
+            [waiter.auth.oauth :refer :all]
+            [waiter.cookie-support :as cs]
+            [waiter.utils :as utils])
+  (:import (clojure.lang ExceptionInfo)
+           org.eclipse.jetty.util.UrlEncoded))
+
+(deftest test-random-str
+  (is (random-str)))
+
+(deftest test-provider-list-handler
+  (testing "get"
+    (let [{:keys [body headers]} (provider-list-handler
+                                   {:github (map->GitHubOAuthProvider {})
+                                    :google (map->GoogleOAuthProvider {})}
+                                   {:request-method :get})]
+      (is body)
+      (is (str/includes? body "GitHub"))
```

Comment:
Instead of `GitHub` or `Google`, should we invoke `display-name` on the providers?